sklearn/ensemble/_hist_gradient_boosting/splitting.pyx

```diff
@@ -433,27 +462,46 @@ cdef class Splitter:
                 # See algo 3 from the XGBoost paper
                 # https://arxiv.org/abs/1603.02754
```

Comment:
this whole comment block should be put below now. Maybe with an additional comment that for categorical features, missing values don't need 2 scans because they are treated as a native category